07-bootstrap-validation.md

```diff
@@ -26,6 +26,42 @@ GitOps allows a team to author Kubernetes manifest files, persist them in their
+
+   > Once a new update is available, it can be applied manually for the greatest degree of control by placing requests against the Azure control plane. Alternatevely, ops team could opt-in to automatically update to the latest available version by configuring an udpate channel following a desired cadence combining this with a planned maintenance window, one for kubernetes version updates and another one for OS level upgrades. AKS provides with two configurable different auto-upgrade channels dedicated to the oforementioned update types. For more information, please refer to  [Upgrade options for Azure Kubernetes Service (AKS) clusters](https://learn.microsoft.com/azure/aks/upgrade-cluster). Nodepools in this AKS cluster span into multiple availability zones, so an important consideration is that automatic updates are conducted based on a best-effort zone balancing in node groups. Nodes Max Surge and Pod Disruption Budget are configured in this baseline to prevent from unbalanced zones increasing the Availabilty. By default clusters nodes are updated one at the time. Max Surge has the ability to increase or reduce the speed of a cluster upgrade. In clusters with 6+ nodes hosting worloads that are sensitive to disruptions, it is recommended up to `33%` surge to achieve a safe upgrade pace. For more information, please take a look at [Customer node surge upgrade](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#customize-node-surge-upgrade). To prevent from disruption, production clusters should be configured with [node draining timeout](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#set-node-drain-timeout-valuei) and [soak time](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli#set-node-soak-time-value) by taking into account the specific charactistics of their workloads.
+
+1. See your maitenance configuration
+
+   ```bash
+   az aks maintenanceconfiguration list --cluster-name $AKS_CLUSTER_NAME -g rg-bu0001a0008
+   ```
+
+   > It is recommended to be prescribed about when upgrades should occur. In case the maitenance windows overlap, AKS decides the running order. Leave at least 24h between maintance window configurations while it will depends based the number of nodes of a particular cluster and the time to upgrade being configured. The OS level updates maintenance window is scheduled to weekly cadence since the OS channel is configured with `NodeImage` where a new node image gets shipped every week. Consider changing this to daily if you opt in for the `SecurityPatch` channel. To stay on top of the kubernetes N-2 version, a monthly cadence is just fine while this cluster is configured every two weeks to keep it updated more regularly. Performing maintenance operations are considered best-effort only and aren't guaranteed to occur within a specified window. While not strictly recommended, for greater level of control consider updating a cluster manually.
```

Comment:
```suggestion
> When managing an Azure Kubernetes Service (AKS) cluster, it’s crucial to plan your upgrades thoughtfully. Here are some recommendations to consider:
>
> Mindful Timing for Upgrades:
> - Be mindful of when upgrades should occur. If you have overlapping maintenance windows, AKS will determine the running order.
> - To avoid conflicts, leave at least 24 hours between maintenance window configurations. The timing will depend on the number of nodes in your specific cluster and the duration required for upgrades.
>
> OS-Level Updates:
> - By default, the OS-level updates maintenance window is scheduled on a weekly cadence. This is because the OS channel is configured with `NodeImage`, where a new node image is shipped every week.
> - If you choose the `SecurityPatch` channel, consider changing the maintenance window to daily for more frequent updates.
>
> Kubernetes Version Management:
> - To stay current with the latest Kubernetes version, a monthly cadence is generally sufficient. However, you can adjust this based on your specific needs.
> - For more regular updates, configure your cluster to upgrade every two weeks.
>
> Maintenance Operations:
> - Keep in mind that performing maintenance operations is considered best-effort. They are not guaranteed to occur within a specific window.
> - While it’s not strictly recommended, if you require greater control, consider manually updating your cluster.
>
> Remember that these guidelines provide flexibility, allowing you to strike a balance between timely updates and operational control. Choose the approach that aligns best with your organization’s requirements.
```